libcxx/test/std/utilities/function.objects/func.bind.partial/bind_back.pass.cpp

```diff
@@ -0,0 +1,356 @@
+      static_assert(std::is_invocable_v<X>);
+      static_assert(!std::is_invocable_v<X const>);
+    }
+  }
+
+  // Some examples by Tim Song
+  {
+    {
+      struct T {};
+      struct F {
+        void operator()(T&&) const&;
+        void operator()(T&&) && = delete;
+      };
+      using X = decltype(std::bind_back(F{}));
+      static_assert(!std::is_invocable_v<X, T>);
+    }
+
+    {
+      struct T {};
+      struct F {
+        void operator()(T const&) const;
+        void operator()(T&&) const = delete;
+      };
+      using X = decltype(std::bind_back(F{}, T{}));
+      static_assert(!std::is_invocable_v<X>);
+    }
+  }
+
+  // Test properties of the constructor of the unspecified-type returned by bind_back.
+  {
+    {
+      MoveOnlyCallable<bool> value(true);
+      auto ret = std::bind_back(std::move(value), 1);
+      assert(ret());
+      assert(ret(1, 2, 3));
+
+      auto ret1 = std::move(ret);
+      assert(!ret());
+      assert(ret1());
+      assert(ret1(1, 2, 3));
+
+      using RetT = decltype(ret);
+      static_assert(std::is_move_constructible<RetT>::value);
+      static_assert(!std::is_copy_constructible<RetT>::value);
+      static_assert(!std::is_move_assignable<RetT>::value);
+      static_assert(!std::is_copy_assignable<RetT>::value);
+    }
+    {
+      CopyCallable<bool> value(true);
+      auto ret = std::bind_back(value, 1);
+      assert(ret());
+      assert(ret(1, 2, 3));
+
+      auto ret1 = std::move(ret);
+      assert(ret1());
+      assert(ret1(1, 2, 3));
+
+      auto ret2 = std::bind_back(std::move(value), 1);
+      assert(!ret());
+      assert(ret2());
+      assert(ret2(1, 2, 3));
+
+      using RetT = decltype(ret);
+      static_assert(std::is_move_constructible<RetT>::value);
+      static_assert(std::is_copy_constructible<RetT>::value);
+      static_assert(!std::is_move_assignable<RetT>::value);
+      static_assert(!std::is_copy_assignable<RetT>::value);
+    }
+    {
+      CopyAssignableWrapper value(true);
+      using RetT = decltype(std::bind_back(value, 1));
+
+      static_assert(std::is_move_constructible<RetT>::value);
+      static_assert(std::is_copy_constructible<RetT>::value);
+      static_assert(std::is_move_assignable<RetT>::value);
+      static_assert(std::is_copy_assignable<RetT>::value);
+    }
+    {
+      MoveAssignableWrapper value(true);
+      using RetT = decltype(std::bind_back(std::move(value), 1));
+
+      static_assert(std::is_move_constructible<RetT>::value);
+      static_assert(!std::is_copy_constructible<RetT>::value);
+      static_assert(std::is_move_assignable<RetT>::value);
+      static_assert(!std::is_copy_assignable<RetT>::value);
+    }
+  }
+
+  // Make sure bind_back is SFINAE friendly
+  {
+    static_assert(!std::is_constructible_v<NotCopyMove, NotCopyMove&>);
+    static_assert(!std::is_move_constructible_v<NotCopyMove>);
+    static_assert(!back_bindable<NotCopyMove>);
+    static_assert(!back_bindable<NotCopyMove&>);
```

Comment:
Perhaps we should use `LIBCPP_STATIC_ASSERT` for libc++-specific SFINAE-friendliness, since the standard uses _Mandates_ for unsuitable parameter types ([[func.bind.partial]/2](https://eel.is/c++draft/func.bind.partial#2)).

Ditto for other occurences of `back_bindable` and pre-existing test cases for `bind_front`.

```suggestion
    LIBCPP_STATIC_ASSERT(!back_bindable<NotCopyMove>);
    LIBCPP_STATIC_ASSERT(!back_bindable<NotCopyMove&>);
```